### PR TITLE
feat: Closes #6956 - Adding confluent health checks

### LINKED ISSUE #6956: Feat: confluent Health Check
Summary
Adding a simple health check for the confluent CR

Motivation
Easy visibility for confluent state in the argo UI and the ability to incorporate this CR in sync waves

Proposal
The standard way for CR health checks.

### PR DESCRIPTION
**Pull request by ThoTischner**
Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [ x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [ x] The title of the PR states what changed and the related issues number (used for the release note).
* [x ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [x ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [x ] Does this PR require documentation updates?
* [ x] I've updated documentation as required by this PR.
* [x ] Optional. My organization is added to USERS.md.
* [x ] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [x ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).

### CODE REVIEW

File: resource_customizations/platform.confluent.io/Connect/health.lua [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    end
+    if obj.status.phase == "PROVISIONING" then
+      hs.status = "Progressing"
+      hs.message = "Connect provisioning"
+      return hs
+    end
+  end
+end
+
+hs.status = "Unknown"
```
Comment (jannfis) [inline resource_customizations/platform.confluent.io/Connect/health.lua:17]: We never return `Unknown` health state in checks, but instead use `Progressing`

File: resource_customizations/platform.confluent.io/ControlCenter/health.lua [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    end
+    if obj.status.phase == "PROVISIONING" then
+      hs.status = "Progressing"
+      hs.message = "ControlCenter provisioning"
+      return hs
+    end
+  end
+end
+
+hs.status = "Unknown"
```
Comment (jannfis) [inline resource_customizations/platform.confluent.io/ControlCenter/health.lua:17]: Please change to `Progressing`

File: resource_customizations/platform.confluent.io/Kafka/health.lua [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    end
+    if obj.status.phase == "PROVISIONING" then
+      hs.status = "Progressing"
+      hs.message = "Kafka provisioning"
+      return hs
+    end
+  end
+end
+
+hs.status = "Unknown"
```
Comment (jannfis) [inline resource_customizations/platform.confluent.io/Kafka/health.lua:17]: Please change to `Progressing`

File: resource_customizations/platform.confluent.io/KsqlDB/health.lua [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    end
+    if obj.status.phase == "PROVISIONING" then
+      hs.status = "Progressing"
+      hs.message = "KsqlDB provisioning"
+      return hs
+    end
+  end
+end
+
+hs.status = "Unknown"
```
Comment (jannfis) [inline resource_customizations/platform.confluent.io/KsqlDB/health.lua:17]: Please change to `Progressing`

File: resource_customizations/platform.confluent.io/SchemaRegistry/health.lua [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+    end
+    if obj.status.phase == "PROVISIONING" then
+      hs.status = "Progressing"
+      hs.message = "SchemaRegistry provisioning"
+      return hs
+    end
+  end
+end
+
+hs.status = "Unknown"
```
Comment (jannfis) [inline resource_customizations/platform.confluent.io/SchemaRegistry/health.lua:17]: Please change to `Progressing`

File: resource_customizations/platform.confluent.io/Zookeeper/health.lua [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+    end
+    if obj.status.phase == "PROVISIONING" then
+      hs.status = "Progressing"
+      hs.message = "Zookeeper provisioning"
+      return hs
+    end
+  end
+end
+
+hs.status = "Unknown"
```
Comment (jannfis) [inline resource_customizations/platform.confluent.io/Zookeeper/health.lua:17]: Please change to `Progressing`

**Review by jannfis — CHANGES_REQUESTED** [review]
Hey @ThoTischner , thank you for this PR.

I have found some minor issues that must be changed, please see below.

File: resource_customizations/platform.confluent.io/Zookeeper/health.lua (hunk 6 above)
Comment (ThoTischner) [reply resource_customizations/platform.confluent.io/Zookeeper/health.lua]: Thanks, Fixed.

File: resource_customizations/platform.confluent.io/SchemaRegistry/health.lua (hunk 5 above)
Comment (ThoTischner) [reply resource_customizations/platform.confluent.io/SchemaRegistry/health.lua]: Thanks, Fixed.

File: resource_customizations/platform.confluent.io/KsqlDB/health.lua (hunk 4 above)
Comment (ThoTischner) [reply resource_customizations/platform.confluent.io/KsqlDB/health.lua]: Thanks, Fixed.

File: resource_customizations/platform.confluent.io/Kafka/health.lua (hunk 3 above)
Comment (ThoTischner) [reply resource_customizations/platform.confluent.io/Kafka/health.lua]: Thanks, Fixed.

File: resource_customizations/platform.confluent.io/ControlCenter/health.lua (hunk 2 above)
Comment (ThoTischner) [reply resource_customizations/platform.confluent.io/ControlCenter/health.lua]: Thanks, Fixed.

File: resource_customizations/platform.confluent.io/Connect/health.lua (hunk 1 above)
Comment (ThoTischner) [reply resource_customizations/platform.confluent.io/Connect/health.lua]: Thanks, Fixed.

**Comment by ishitasequeira** [comment]
@ThoTischner can you change the PR title/ commit message for the semantic check to pass? You could follow https://github.com/zeke/semantic-pull-requests

**Comment by ThoTischner** [comment]
> @ThoTischner can you change the PR title/ commit message for the semantic check to pass? You could follow https://github.com/zeke/semantic-pull-requests

Done, thanks :)

**Review by ishitasequeira — COMMENTED** [review]
LGTM. @jannfis could you review the same?

**Review by jannfis — APPROVED** [review]
LGTM, thanks @ThoTischner and also thanks @ishitasequeira for doing a review!

**Pull request merged**